### ISSUE TITLE
Torchcvnn

### ISSUE THREAD

**Issue by narausch** [pytorch-fdn/ecosystem#16]
### Contact emails

[jeremy.fix@centralesupelec.fr](mailto:jeremy.fix@centralesupelec.fr)

### Project summary

The goal of the project is to extend the usability of complex valued neural networks.

### Project description

The goal of the project is to extend the usability of complex valued neural networks.

It provides :
- datasets constructors with connectors to read standard radar datasets. The radar context is one application context where the data are complex by nature and we provide connectors to read data in the SLC and ALOS2 formats
- complex valued activations which are more than just the real valued activations applied independently to both the real and imaginary components
- complex valued batchnormalization as suggested in Trabelsi et al 2017

And other layers that fail for now with upstream pytorch with tensors of complex value datatype although nothing really enforces the tensors to be real valued. For example, Dropout should be mask the tensor whichever the datatype it hosts but applying Dropout on complex valued tensors fails for now with pytorch.

### Are there any other projects in the PyTorch Ecosystem similar to yours?  If, yes, what are they?

Yes, I found the following projects:
- complexPyTorch : https://github.com/wavefrontshaping/complexPyTorch 

- cplxmodule : https://github.com/ivannz/cplxmodule 
 
Repo URL: https://github.com/jeremyfix/torchcvnn 

### Project repo URL

 https://github.com/jeremyfix/torchcvnn

### Project license

MIT

### GitHub handles of the project maintainer(s)

 jeremyfix

### Is there a corporate or academic entity backing this project?  If so, please provide the name and URL of the entity.

Yes, I do belong to CentraleSupelec, member of the LORIA laboratory, which are public academic entities.

### Website URL

https://github.com/jeremyfix/torchcvnn

### Documentation

I provide some example codes you can see in the documentation and the "examples" subdirectory of the repository :

For example, classifying MNIST from their fast fourier transform representation is documented here : https://jeremyfix.github.io/torchcvnn/reference/examples/mnist/ 

https://jeremyfix.github.io/torchcvnn/reference/examples/mnist/ 

### How do you build and test the project today (continuous integration)?  Please describe.

For now, we provide the CI/CD :
- builds and deploy the release to pypi
- run unit tests
- builds and deploy the documentation

Several unit tests are implemented in https://github.com/jeremyfix/torchcvnn/tree/main/tests

### Version of PyTorch

I provide some example codes you can see in the documentation and the "examples" subdirectory of the repository :

For example, classifying MNIST from their fast fourier transform representation is documented here : https://jeremyfix.github.io/torchcvnn/reference/examples/mnist/ 

https://jeremyfix.github.io/torchcvnn/reference/examples/mnist/ 

### Components of PyTorch

NA

### How long do you expect to maintain the project?

A PhD thesis started last year on the topic of complex valued generative networks and he is using it, which motivates me maintaining it.

**Comment by narausch** [comment]
The Ecosystem review team believes this project is still too early to join the system.  Joe Spisak proposed the idea of building a community to support and enhance the project. Luca Antiga recommended creating comprehensive documentation to support community engagement and usability. Next steps are to support a blog post and other opportunities for Jeremy to grow this work.

**Comment by narausch** [comment]
This was reviewed in the November TAC meeting by the Ecosystem working group.
